core/src/test/java/org/elasticsearch/threadpool/ScalingThreadPoolTests.java

```diff
@@ -177,7 +177,8 @@ public void testScalingThreadPoolThreadsAreTerminatedAfterKeepAlive() throws Int
                     }
                 });
             }
-            assertThat(stats(threadPool, threadPoolName).getThreads(), equalTo(128));
+            int threads = stats(threadPool, threadPoolName).getThreads();
+            assertEquals(128, threads);
```

Comment:
It seems like something the compiler is broken here. Is there a compiler bug open for this? Or is there a spec change that makes the change here necessary and the compiler is not broken?